tests/unit/test_client.py

```diff
@@ -0,0 +1,130 @@
+import pytest
+from mock import patch, Mock
+
+from plaid.client import (
+    Client,
+    inject_credentials,
+    inject_url,
+    store_access_token
+)
+from plaid.errors import UnauthorizedError
+
+
+def test_inject_credentials_decorator():
+    class TestClass(object):
+        def __init__(self):
+            self.client_id = 'bob'
+            self.secret = 'my_secret'
+            self.access_token = 'access_me'
+
+        @inject_credentials
+        def some_func(self, credentials, url):
+            return [credentials, url]
+
+    plaid_url = 'http://plaid.com'
+
+    obj = TestClass()
+    creds, url = obj.some_func(plaid_url)
+    assert url == plaid_url
+    assert creds == {
+        'client_id': 'bob',
+        'secret': 'my_secret',
+        'access_token': 'access_me'
+    }
+
+
+def test_failed_inject_credentials_decorator():
+    class TestClass(object):
+        def __init__(self):
+            self.client_id = 'bob'
+            self.secret = 'my_secret'
+
+        @inject_credentials
+        def some_func(self, credentials, url):
+            return [credentials, url]
+
+    plaid_url = 'http://plaid.com'
+
+    obj = TestClass()
+    try:
+        creds, url = obj.some_func(plaid_url)
+    except UnauthorizedError, e:
+        assert(e.code == 1000)
+        assert(e.message == 'some_func requires `access_token`')
+    else:
+        assert False
+
+
+def test_inject_url():
+    class TestClass(object):
+        base_url = 'http://plaid.com'
+        ENDPOINTS = {
+            'auth': '/auth',
+        }
+
+        @inject_url('auth')
+        def some_func(self, url, extra):
+            return [url, extra]
+
+    obj = TestClass()
+    url, extra = obj.some_func(5)
+    assert extra == 5
+    assert url == 'http://plaid.com/auth'
+
+
+def test_store_access_token():
+    class Response(object):
+        status_code = 200
+        content = '{"access_token": 5}'
+
+
+    class TestClass(object):
+        def __init__(self):
+            self.access_token = None
+
+        @store_access_token
+        def some_func(self):
+            return Response()
+
+    obj = TestClass()
+    response = obj.some_func()
+    assert response.content == '{"access_token": 5}'
```

Comment:
This test is too restrictive: it will fail if the JSON is formatted differently, such as if it doesn't have a space after the colon (`:`). What we really want to assert is `parse(response.content) == {'access_token': 5}`.